### PR TITLE
[Bugfix] Fixes code line highlighting for tutors and wrongly invoked method for students

### PR DESCRIPTION
**Pull request by fde312**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) locally
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Server: I added multiple integration tests (Spring) related to the features (with a high test coverage)
- [x] Server: I added `@PreAuthorize` and check the course groups for all new REST Calls (security)
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Motivation and Context

This PR solves two issue found during retesting of the manual assessment of programming exercises. Issue 1: when a tutor started an assessment the code line changes were not correctly highlighted as an error was thrown. Issue 2: When a student wanted to view his code in the online editor after receiving a manual assessment, he could not select a file as an error was thrown.

### Description

- Added new REST Endpoint to just fetch information about the template and solution participation of a exercise (issue1)
- Extended the check so that `file-changes` call is only executed during the assessment (issue2)

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Administration
3. Create a programming exercise, participate and do manual assessments or use a existing programming exercise with manual assessments
4. Log in as tutor and check if the code line highlighting is correct and assess the submission
5. Log in as student and check if you can select files and that the according inline feedback is displayed

### Test Coverage

Nothing changed.

### Screenshots

Student View:
image

Tutor View:
image

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java [hunk 1]
```diff
@@ -799,6 +799,32 @@ else if (!authCheckService.isAtLeastTeachingAssistantForExercise(programmingExer
         }
     }
 
+    /**
+     * GET /programming-exercises/:exerciseId/with-template-and-solution-participation
+     *
+     * @param exerciseId the id of the programmingExercise to retrieve
+     * @return the ResponseEntity with status 200 (OK) and the programming exercise with template and solution participation, or with status 404 (Not Found)
+     */
+    @GetMapping(Endpoints.PROGRAMMING_EXERCISE_WITH_TEMPLATE_AND_SOLUTION_PARTICIPATION)
```
Comment (fde312) [inline src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java:808]: @krusche We could also use the REST Call which is above this one and adapt the checks to also allow tutors to use it.

**Review by airelawaleria — APPROVED** [review]
Worked on ts1. 
The only thing (maybe unrelated to this pr) i noticed is:
If as a tutor I create an inline comment (e.g. line 5), but don't explicitly save it, then go to another file and try to create an inline comment in the same line (line 5 in another file), I see the unsaved comment from the previous file. Is it intended?

**Review by LudwigWeide — APPROVED** [review]
Approve with same issue as @airelawaleria .

**Review by JohannesWeiss — APPROVED** [review]
Tested on ts1 and worked as expected wit exception of the aforementioned issue.

**Review by simonzilker — APPROVED** [review]
What @airelawaleria said

**Review by marvinchimeka — APPROVED** [review]
tested in testing session. worked as intended. Got the same issue as @airelawaleria.

File: src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java (hunk 1 above)
Comment (jpbernius) [reply src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java:808]: I guess this is different enough to keep it as a separate endpoint.

**Review by jpbernius — APPROVED** [review]

File: src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java (hunk 1 above)
Comment (krusche) [reply src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java:808]: The one above also includes the latest result which is probably not needed.
I think we can use a new one for now to fix the bugs.

**Review by krusche — APPROVED** [review]

**Review by EneaGore — APPROVED** [review]

**Pull request merged**